ibtk/src/solvers/impls/PETScLevelSolver.cpp

```diff
@@ -307,6 +314,19 @@ void PETScLevelSolver::deallocateSolverState()
     return;
 } // deallocateSolverState
 
+void PETScLevelSolver::addLinearOperator(Mat& op, MatStructure nonzero_pattern)
+{
+#if !defined(NDEBUG)
+	TBOX_ASSERT(!d_is_initialized);
+	TBOX_ASSERT(op);
+#endif
+
+	d_petsc_extern_mat = op;
+	d_extern_mat_nz_pattern = nonzero_pattern;
+
+	return;
+}// addLinearOperator
+
 /////////////////////////////// PROTECTED ////////////////////////////////////
 
 void PETScLevelSolver::init(Pointer<Database> input_db, const std::string& default_options_prefix)
```

Comment:
Hmmm.. that routine is actually adding another linear operator to the one the derived class is building...'add' sounds more apt, doesn't it?